src/Slack/SlackMessage.php

```diff
@@ -273,17 +286,21 @@ public function broadcastReply(?bool $broadcastReply = true): self
      */
     public function toArray(): array
     {
-        if (empty($this->blocks) && $this->text === null) {
+        if (empty($this->blocks) && is_null($this->blockBuilder) && $this->text === null) {
             throw new LogicException('Slack messages must contain at least a text message or block.');
         }
 
-        if (count($this->blocks) > 50) {
+        $builderBlocks = is_null($this->blockBuilder) ? [] : $this->blockBuilder->toArray();
+
+        if (count($this->blocks) > 50 || count($builderBlocks) > 50) {
             throw new LogicException('Slack messages can only contain up to 50 blocks.');
         }
 
         $optionalFields = array_filter([
             'text' => $this->text,
-            'blocks' => ! empty($this->blocks) ? array_map(fn (BlockContract $block) => $block->toArray(), $this->blocks) : null,
+            'blocks' => is_null($this->blockBuilder)
+                ? (! empty($this->blocks) ? array_map(fn (BlockContract $block) => $block->toArray(), $this->blocks) : null)
+                : $builderBlocks,
```

Comment:
Assuming we'll stick to this approach instead of relaxing the strictness of the $blocks property, it would be cleaner if we create a check and variable for this instead of nesting conditionals, as that way the code is more readable and maintainable.